webapp/src/components/ExchangePage/index.jsx

```diff
@@ -369,112 +438,239 @@ function getMarketRate(
-          )
-          const { result } = await res.json()
-          // @TODO: UAF - please change it, shame on you Nacho
-          // Check if the extra data is related to an order
-          const indexOfTransfer = result ? result.input.indexOf(TRANSFER_SELECTOR) : -1
-          if (indexOfTransfer !== -1 && result.input.length > ORDER_BYTES_LENGTH) {
-            const orderData = `0x${result.input.substr(
-              indexOfTransfer + TRANSFER_TX_LENGTH + TX_PADDED_BYTES_BOILERPLATE,
-              ORDER_BYTES_LENGTH
-            )}`
-            try {
-              const order = await decodeOrder(uniswapEXContract, orderData)
-              if (!order) {
-                if (ordersAdded[orderData] >= 0) {
-                  delete orders[ordersAdded[orderData]]
-                  delete ordersAdded[orderData]
-                }
-                continue
-              }
-              const vault = await uniswapEXContract.vaultOfOrder(...Object.values(order))
-              const amount = await new Promise((resolve, reject) =>
-                window.web3.eth.call(
-                  {
-                    to: order.fromToken,
-                    data: `${BALANCE_SELECTOR}000000000000000000000000${vault.replace('0x', '')}`
-                  },
-                  (error, amount) => {
-                    if (error) {
-                      reject(error)
-                    }
-                    resolve(amount)
-                  }
-                )
-              )
-              if (order && ordersAdded[orderData] === undefined) {
-                orders.push({ ...order, amount })
-                ordersAdded[orderData] = orders.length - 1
-              }
-            } catch (e) {
-              console.warn(`Error decoding order: ${orderData}`)
-            }
-          }
+function findOrders(data) {
+  const orders = []
+  const transfers = data.split('a9059cbb').slice(1)
+
+  for (const transfer of transfers) {
+    if (transfer.length >= 704) {
+      const order = transfer.slice(256, 706)
+      // Uniswap orders have a pk and address embeded on the tx
+      // we look for those
+      const pk = `0x${order.slice(320, 320 + 64)}`
+      const expectedAddress = `0x${order.slice(408, 408 + 40)}`
+      if (new ethers.Wallet(pk).address.toLowerCase() === expectedAddress) {
+        orders.push(`0x${order.slice(1)}`)
+      }
+    }
+  }
+
+  return orders
+}
+
+async function backfillOrders(account, onUpdate = _ => {}) {
+  const reviewedTxs = new Set()
+
+  const targetBlock = await readWeb3.eth.getBlockNumber()
+  const last = getLastBackfill(account)
+  console.info(`Running backfill for ${account} - ${last}/${targetBlock}`)
+
+  const logs = await readWeb3.eth.getPastLogs({
+    fromBlock: last,
+    toBlock: targetBlock,
+    topics: [
+      '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef', // Transfer topic
+      `0x000000000000000000000000${account.replace('0x', '')}` // Account topic
```

Comment:
Should we need to check if the account is not defined here?

Reply:
Same, is checked on line 546, same file